### ISSUE TITLE
`Stabilizer.expectation_value()` does not give the correct result

### ISSUE THREAD

**Issue by AlaricCheng** [Qiskit/qiskit-terra#7441]
### Environment

- **Qiskit Terra version**: 0.19.1
- **Python version**: 3.9.5
- **Operating system**: Win 10

### What is happening?

I wrote a script for generating random Clifford circuit with respect to certain layout. Then I convert it to `StabilizerState` and compute the expectation value of its stabilizers with `StabilizerState.expectation_value()`, which should be one. The stabilizers are obtained with `StabilizerState.clifford.to_dict()`. But some expectation values are zero, instead of one. 

### How can we reproduce the issue?

The following code can be used to reproduce the issue.
```python
from qiskit.quantum_info import Pauli, random_clifford, StabilizerState
from qiskit import QuantumCircuit
from numpy.random import RandomState

n_qubits = 3
circ = QuantumCircuit(n_qubits)

compute_seed = RandomState(0).choice(10, size = (1, n_qubits))

for i in range(n_qubits): 
    gate = random_clifford(1, compute_seed[0, i])
    circ.compose(gate.to_circuit(), [i], inplace = True)

print(circ)
s = StabilizerState(circ)
stab_gen = s.clifford.to_dict()["stabilizer"]
print(stab_gen)

for i in range(n_qubits):
    print(s.expectation_value(Pauli(stab_gen[i]))) # should be 1
```
The printed result of last line is 
```python
1
1
0
```

### What should happen?

The values should all be one. Actually, if we run the following code
```python
stab_sim = Aer.get_backend("aer_simulator_stabilizer")
for i in range(3):
    circ.save_expectation_value(Pauli(stab_gen[i]), circ.qubits)
    result = execute(circ, stab_sim).result()
    print(result.data(circ))
    circ.data.pop(-1)
```
Then, the printed results are all `{'expectation_value': 1.0}`.

**Comment by ShellyGarion** [comment]
I think that the bug is when the Pauli operator has a negative sign, continuing the above example:

```
s = StabilizerState(circ)
vec = Statevector(circ)
print (s)
print (s.expectation_value(Pauli("-YII")))
print (vec.expectation_value(Pauli("-YII")))
```
gives:
```
StabilizerState(StabilizerTable: ['-IIY', '+IXI', '-YII'])
0
1
```

**Comment by ShellyGarion** [comment]
@AlaricCheng - thank you very much for finding this bug and opening an issue.
I fixed it in PR #7460 , and added a test based on your example above.
